lib/install/actions.js

```diff
@@ -46,7 +46,8 @@ Object.keys(actions).forEach(function (actionName) {
       if (!pkg.commit) pkg.commit = []
       pkg.commit.push(action.commit)
     }
-    if (pkg.knownInstallable) {
+
+    if (pkg.knownInstallable || pkg.package._optional) {
```

Comment:
Ok! I've updated the code and added a test for the case.  Here is the minimum case to reproduce the original issue: https://github.com/watilde/npm-issue-14042